kytos/core/tcp_server.py

```diff
@@ -44,113 +40,108 @@ def __init__(self, server_address, RequestHandlerClass,
         """
         super().__init__(server_address, RequestHandlerClass,
                          bind_and_activate=False)
-        # Registering the register_event 'endpoint' on the server to be
-        #   accessed on the RequestHandler
         self.controller = controller
-        # self.controller_put_raw_event = controller_put_raw_event
 
     def serve_forever(self, poll_interval=0.5):
-        """Handle requests until an explicit shutdown() request."""
+        """Handle requests until an explicit shutdown() is called."""
+        SOCKET_BIND_WAIT_TIME = 0.5
+        SOCKET_BIND_N_TRIES = 2
+        # Tries to bind the socket n times, it may still be
+        # in WAIT state from previous connection.
+        for i in range(SOCKET_BIND_N_TRIES):
```

Comment:
@abaruchi @renanrodrigo @diraol @beraldoleal,
I've moved my solution to another branch in case someone wants to consider it: https://github.com/erickvermot/kytos/tree/socket_bind_fix_wait
It retries N times waiting X seconds between retries, and only forces the bind if the retries fails.
This gives at least some time (N*X) before forcing the bind, and may help prevent packets from an old connection and possibly from another switch to eventually be received by the new connection.